open_lm/train.py

```diff
@@ -268,13 +312,20 @@ def train_one_epoch(model, data, loss, epoch, step, optimizer, scaler, scheduler
 
             # gathered_loss = [torch.zeros_like(total_loss) for _ in range(args.world_size)]
             # torch.distributed.all_gather(gathered_loss, total_loss)
+
             # losses_m.update(sum(gathered_loss).item() / args.world_size, batch_size * args.world_size)
-            losses_m.update(global_loss_tensor.item(), batch_size)
+            if args.moe_freq > 0:
+                losses_m.update(global_loss_tensor.item() - total_load_balancing_loss.item(), batch_size)
+                load_balancing_losses_m.update(total_load_balancing_loss.item(), batch_size)
+            else:
+                losses_m.update(global_loss_tensor.item(), batch_size)
             samples_per_second = inputs.numel() * args.world_size / batch_time_m.val
             samples_per_second_per_gpu = inputs.numel() / batch_time_m.val
+            loss_str = f"Loss: {losses_m.avg:.3f}"
+            loss_str += f" LB-Loss: {load_balancing_losses_m.avg:.3f}" if args.moe_freq > 0 else ""
```

Comment:
maybe guard this with a args.moe_freq > 0 ? also might be good do do a open_lm_160m run with this branch to make sure nothing in non-moe codepath affected